test-common/src/main/java/de/bwaldvogel/mongo/backend/AbstractBackendTest.java

```diff
@@ -3088,6 +3094,31 @@ public void testRenameField() throws Exception {
 
         collection.updateOne(json("_id: 1"), json("$rename: {'bar2': 'foo', foo2: 'bar'}"));
         assertThat(collection.find().first()).isEqualTo(json("_id: 1, bar: 'x', foo: 'y'"));
+
+        collection.updateOne(json("_id: 1"), json("$rename: {'bar': 'bar2', 'missing': 'foo'}"));
+        assertThat(collection.find().first()).isEqualTo(json("_id: 1, bar2: 'x', foo: 'y'"));
+    }
+
+    @Test
+    public void testRenameField_embeddedDocument() {
+        Document object = json("_id: 1, foo: { a: 1, b: 2 }, bar: { c: 3, d: 4 }}");
+        collection.insertOne(object);
+
+        collection.updateOne(json("_id: 1"), json("$rename: {'foo.a': 'foo.z', 'bar.c': 'bar.x'}"));
+        assertThat(collection.find().first()).isEqualTo(json("_id: 1, foo: { z: 1, b: 2 }, bar: { x: 3, d: 4 }}"));
+
+        collection.updateOne(json("_id: 1"), json("$rename: {'foo.z': 'foo.a', 'bar.a': 'bar.b'}"));
+        assertThat(collection.find().first()).isEqualTo(json("_id: 1, foo: { a: 1, b: 2 }, bar: { x: 3, d: 4 }}"));
+
+        collection.updateOne(json("_id: 1"), json("$rename: {'missing.a': 'missing.b'}"));
+        assertThat(collection.find().first()).isEqualTo(json("_id: 1, foo: { a: 1, b: 2 }, bar: { x: 3, d: 4 }}"));
+
+        collection.updateOne(json("_id: 1"), json("$rename: {'foo.a': 'a', 'bar.x': 'bar.c'}"));
+        assertThat(collection.find().first()).isEqualTo(json("_id: 1, foo: { b: 2 }, bar: { c: 3, d: 4 }, a: 1}"));
+
+        assertThatExceptionOfType(Exception.class).isThrownBy(
```

Comment:
Is there a better way to do this? Constructing a full error message for a MongoServerError is insanely difficult for this particular case.

Reply:
We should at least make sure the right kind of exception is thrown.
If you expect an exception of type `Exception`, it could even be something like a `NullPointerException` that shouldn’t be thrown here.

Reply:
Changed this to `MongoWriteException`